include/deal.II/base/polynomial_space.h

```diff
@@ -141,7 +141,9 @@ class PolynomialSpace
   void compute (const Point<dim>            &unit_point,
                 std::vector<double>         &values,
                 std::vector<Tensor<1,dim> > &grads,
-                std::vector<Tensor<2,dim> > &grad_grads) const;
+                std::vector<Tensor<2,dim> > &grad_grads = Utilities::EmptyVectorGenerator<Tensor<2,dim> >().get_reference(),
+                std::vector<Tensor<3,dim> > &grad_hessians = Utilities::EmptyVectorGenerator<Tensor<3,dim> >().get_reference(),
+                std::vector<Tensor<4,dim> > &grad_hessian_grads = Utilities::EmptyVectorGenerator<Tensor<4,dim> >().get_reference()) const;
```

Comment:
Yes that's true. The long text to get the default empty reference is not great within the function signatures either.

How about an intermediate solution where the default empty vector variables are defined as const private member variables within the classes that own compute? Something like:

private:
...
const std::vector<Tensor<3,dim> > empty_vector_of_tensor_3s;
const std::vector<Tensor<4,dim> > empty_vector_of_tensor_4s;
....

at least then the changes are restricted to classes where the functions are defined, the signature becomes much more readable, and backwards compatibility is preserved.